stock_picking_mass_action/wizard/mass_action.py

```diff
@@ -0,0 +1,110 @@
+    _description = 'Stock Picking Mass Action'
+
+    @api.model
+    def _default_check_availability(self):
+        return self.env.context.get('check_availability', False)
+
+    @api.model
+    def _default_force_availability(self):
+        return self.env.context.get('force_availability', False)
+
+    @api.model
+    def _default_transfer(self):
+        return self.env.context.get('transfer', False)
+
+    def _default_picking_ids(self):
+        return self.env['stock.picking'].browse(
+            self.env.context.get('active_ids'))
+
+    confirm = fields.Boolean(
+        string='Mark as Todo',
+        default=True,
+        help="check this box if you want to mark as Todo the"
+        " selected Pickings.",
+    )
+    check_availability = fields.Boolean(
+        string='Check Availability',
+        default=lambda self: self._default_check_availability(),
+        help="check this box if you want to check the availability of"
+        " the selected Pickings.",
+    )
+    force_availability = fields.Boolean(
+        string='Force Availability',
+        default=lambda self: self._default_force_availability(),
+        help="check this box if you want to force the availability"
+        " of the selected Pickings.",
+    )
+    transfer = fields.Boolean(
+        string='Transfer',
+        default=lambda self: self._default_transfer(),
+        help="check this box if you want to transfer all the selected"
+        " pickings.\n You'll not have the possibility to realize a"
+        " partial transfer.\n If you want  to do that, please do it"
+        " manually on the picking form.",
+    )
+    picking_ids = fields.Many2many(
+        string='Pickings',
+        comodel_name="stock.picking",
+        default=lambda self: self._default_picking_ids(),
+        help="",
+    )
+
+    @api.multi
+    def mass_action(self):
+        self.ensure_one()
+
+        # Get draft pickings and confirm them if asked
+        if self.confirm:
+            draft_picking_lst = self.picking_ids.\
+                filtered(lambda x: x.state == 'draft').\
+                sorted(key=lambda r: r.scheduled_date)
+            draft_picking_lst.action_confirm()
+
+        # Get confirmed pickings
+        confirmed_picking_lst = self.picking_ids.\
+            filtered(lambda x: x.state in [
+                'confirmed',
+                'partially_available',
+            ]).\
+            sorted(key=lambda r: r.scheduled_date)
+
+        # check availability if asked
+        if self.check_availability:
+            pickings_to_check = self.picking_ids.\
+                filtered(lambda x: x.state not in [
+                    'draft',
+                    'cancel',
+                    'done',
+                ]).\
+                sorted(key=lambda r: r.scheduled_date)
+            pickings_to_check.action_assign()
+
+        # Force availability if asked
+        if self.force_availability:
+            confirmed_picking_lst.force_assign()
```

Comment:
There's no `force_assign` method in `12.0` and I can't find an equivalent in the core. Funny, because it was used mainly by the point_of_sale to deliver stock no matter what

Reply:
I suppose the equivalent is to write in `qty_done` on stock.move.line records the forced quantity.

Reply:
Sorry. Changes done.
I was working with my environment on v11 :sweat_smile: